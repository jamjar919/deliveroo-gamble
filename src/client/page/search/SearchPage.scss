@use "sass:map";
@import "../../App.scss";

.container {
  display: flex;
  flex-direction: column;
  height: 100vh;

  padding: 0 20px;
  @media (min-width: map.get($grid-breakpoints, "sm")) {
    padding: 0;
  }
}

.formContainer {
  margin: auto;
  display: flex;
  flex-direction: column;

  @media (min-width: map.get($grid-breakpoints, "lg")) {
    min-width: 575px;
  }
}
