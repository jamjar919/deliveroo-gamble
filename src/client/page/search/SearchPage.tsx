import React from "react";
import {SearchPageForm, SearchPageFormField, SearchPageFormValues} from "./form/SearchPageForm";
import {useGambleContext} from "../../context/GambleContext";

import styles from './SearchPage.scss';

const SearchPage: React.FC = () => {
    const {
        postcodeGamble,
    } = useGambleContext();

    return (
        <div className={styles.container}>
            <div className={styles.formContainer}>
                <SearchPageForm
                    onSubmit={(values: SearchPageFormValues) => {
                        const postcode = values[SearchPageFormField.POSTCODE];
                        const price = Number(values[SearchPageFormField.PRICE_LIMIT]) * 100;
                        const firstItemIsLarge = values[SearchPageFormField.FIRST_ITEM_IS_LARGE];

                        return postcodeGamble(
                            postcode,
                            price,
                            firstItemIsLarge
                        )
                    }}
                />
            </div>
            <div className={styles.footer}>
                <div className={styles.footerItem}>
                    Developed by <a href={"https://thejamespaterson.com"}>James Paterson</a>
                </div>
                <div className={styles.footerItem}>‚ùÇ</div>
                <div className={styles.footerItem}>
                    Not affiliated with Deliveroo or Just Eat
                </div>
            </div>
        </div>
    )
}

export { SearchPage };