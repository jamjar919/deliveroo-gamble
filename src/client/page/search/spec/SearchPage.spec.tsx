import React from "react";
import {SearchPage} from "../SearchPage";
import {validateRender} from "../../../../../test/validateRender";

describe('SearchPage', () => {
    it('renders', () => {
        validateRender(<SearchPage
            onSearch={() => Promise.resolve()}
        />);
    });
});