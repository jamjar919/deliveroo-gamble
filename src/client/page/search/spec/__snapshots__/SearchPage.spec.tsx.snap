// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage renders 1`] = `
<div
  className="container"
>
  <div
    className="formContainer"
  >
    <img
      alt="Deliveroo Logo"
      className="logo"
      src="/deliveroo-logo.png"
    />
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="inputContainer"
      >
        <div
          className="unit"
        >
          Â£
        </div>
        <input
          className="input"
          name="PRICE_LIMIT"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="12.00"
          type="numeric"
          value="12.00"
        />
        <div
          className="buttonContainer"
        >
          <button
            className="button"
            type="submit"
          >
            Gamble!
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
